.travis.yml

```diff
@@ -14,15 +14,19 @@ matrix:
       otp_release: 19.3
     - elixir: 1.7
       otp_release: 21.0
-      env: EXTRA_CHECKS=true
 
-env:
-  - EXTRA_CHECKS=false
+install:
+  - mix local.rebar --force
+  - mix local.hex --force
+  - mix deps.get --only test
+
+before_script:
+  - MIX_ENV=test mix compile --warnings-as-errors
 
 script:
-  - MIX_ENV=test mix deps.compile
-  - ./ci/tests.sh
+  - if [[ "$TRAVIS_ELIXIR_VERSION" == "1.7"* ]]; then mix format --check-formatted; fi
+  - MIX_ENV=test mix coveralls.travis
```

Comment:
Whats the consequence of reporting for all environments?

Reply:
Coveralls.io now appears to handle this well. It lists individual coverage values for each build environment. Previously each environment would race to report, and either the first or last would win (I don’t recall).